### DISCUSSION TITLE
Workflow to release from CI?

### DISCUSSION THREAD

**Discussion by flying-sheep** [release-plz/release-plz#709] [Q&A] [answered]
I have a hybrid Rust crate / Python package: https://github.com/flying-sheep/xdot-rust

I would like to use a workflow where

1. for each commit to `main` that isn’t a release, a tool creates a release PR with changelog and suggested version (release-plz seems to fit the bill perfectly here)
2. when that PR is merged (creating a release commit), CI should

   - create a Git tag and a GitHub release
   - publish the crate to crates.io
   - build the Python package and publish it to PyPI

Can `release-plz release` be run from CI as response to release commits? Then it would be suited for this workflow. Running it locally isn’t a good idea because I also need to build and publish the Python package.

**Comment by marcoieni** [comment, marked as answer]
Hi! Have you read the [docs](https://marcoieni.github.io/release-plz/)? Especially the GitHub action [section](https://marcoieni.github.io/release-plz/github/index.html).
What you described is the main purpose of Release-plz. The fact that you missed it it's a failure in organizing the README/docs on my side.
Do you have any suggestions on how to improve in this? How can I make the docs more clear or more discoverable?

**Reply by flying-sheep** [reply]
Oh, now I see that it’s actually pretty clear. I just took a weird path:

1. I read the README
2. I checked the linked Users and didn’t see them using the `release` command in their CI
3. I read the [CLI docs for `release`](https://marcoieni.github.io/release-plz/usage/release.html)

I totally missed the obvious “[GitHub action](https://marcoieni.github.io/release-plz/github/index.html#github-action)” section, which clearly explains that it runs both commands.

The things I can think of that could maybe have tipped me off would have been:

- Explaining the way it works in the README. I didn’t understand that the action does both things based on what commits it finds. I for some reason assumed that the action people were using just created the PR but didn’t do the `release` step.
- Linking the “GitHub actions” section from the README.
- Naming the chapters sth. like “Manual usage” and “CI Usage: GitHub actions”
- Linking the “GitHub actions” subsection

I think the best thing to do would be to change this part of the README:

> Once the changes are merged to the main branch, you can use `release-plz release` to publish the new versions of the packages.

to

> When changes are merged to the main branch, `release-plz release` will publish the new versions of the packages.
> 
> In the simplest case, just use the [GitHub Action](https://marcoieni.github.io/release-plz/github/index.html) to perform either task: create a PR or publish crate & release.

or similar

**Reply by flying-sheep** [reply]
I think I was also confused because I couldn’t see a way to have it only create a tag and a release (skipping the cargo upload) and instead have a workflow run `on: release` that does the upload for only that release (which I could then use for Python as well)

**Comment by marcoieni** [comment]
Thank you for your feedback, it was really valuable!
In https://github.com/MarcoIeni/release-plz/pull/723/files I changed the readme, taking inspiration from [release-please](https://github.com/googleapis/release-please).
What do you think? :)

**Reply by flying-sheep** [reply]
It’s great! Quite clear.
